rakelib/form526.rake

```diff
@@ -102,8 +54,73 @@ namespace :form526 do
     puts '* Failure Counts for form526 Submission Job (not including uploads/cleanup/etc...) *'
     print_total('Outage Failures: ', outage_errors)
     print_total('Other Failures: ', other_errors)
+  end
+
+  desc 'Get an error report within a given date period. [<start date: yyyy-mm-dd>,<end date: yyyy-mm-dd>]'
+  task :errors, %i[start_date end_date] => [:environment] do |_, args|
+    def print_row(sub_id, p_id, created_at)
+      printf "%-20s %-20s %s\n", sub_id, p_id, created_at
+    end
+
+    def print_errors(errors)
+      errors.each do |k, v|
+        puts k
+        puts '*****************'
+        puts "Unique Participant ID count: #{v[:participant_ids].count}"
+        print_row('submission_id:', 'participant_id:', 'created_at:')
+        v[:submission_ids].each do |submission|
+          print_row(submission[:sub_id], submission[:p_id], submission[:date])
+        end
+        puts '*****************'
+        puts ''
+      end
+    end
+
+    MSGS_REGEX = /key\\"=>\\"(.*?)\\".*?text\\"=>\\"(.*?)\\"/
+
+    start_date = args[:start_date]&.to_date || 30.days.ago.utc
+    end_date = args[:end_date]&.to_date || Time.zone.now.utc
+
+    submissions = Form526Submission.where(
+      'created_at BETWEEN ? AND ?', start_date.beginning_of_day, end_date.end_of_day
+    )
+
+    errors = {}
+
+    submissions.find_each do |s|
+      auth_headers = JSON.parse(s.auth_headers_json)
+      s.form526_job_statuses.each do |j|
+        if j.error_class.present?
```

Comment:
could this be a guard clause? 
```ruby 
next unless j.error_class.present?
```

Reply:
It totally could and it totally will be.